git/operations.go

```diff
@@ -41,11 +42,31 @@ func clone(ctx context.Context, workingDir string, keyRing ssh.KeyRing, repoURL,
 	return repoPath, nil
 }
 
-func commit(ctx context.Context, workingDir, commitMessage string) error {
+func commit(ctx context.Context, workingDir string, commitAction *CommitAction) error {
+	commitAuthor := commitAction.Author
+	// The user in the update.Cause.User must be a valid github user, otherwise the commit fails
+	// Hardcoding of the user to ignore is not ideal. If there is only one to ignore, ie test@test,
+	// then this is just about ok. Otherwise there could be config file with stored users to ignore etc.
+	// Currently the committer is hardcoded to flux-bot for user requested actions.
+	// TODO (potentially): allow the users to set the committer, ie have an additional
+	// update.Cause.Committer field.
```

Comment:
The committer is already supplied as the arguments to the daemon -- `--git-user` and `--git-email` -- and written into the repo config when it's cloned. Unless there's some reason to vary it per commit, I don't think this new code needs to do anything relating to the committer.

Reply:
a)
I  misunderstood our discussion about the custom committer a couple of days. I thought you were suggesting an additional fluxctl option. Understanding corrected.
b)
I  always had the default git-user, Weave Flux, as a commit author and a committer prior to this implementation.  (This poses a question (to me) what git-user and git-email is used elsewhere when starting the daemon. ) My understanding is that non user requested commits are using that (whether the default value or not). The issue 695 talks about having a different commit author/committer for user requested actions. This I interpret that different commits, depending on whether through actions requested by the user, can have a different committer.

Reply:
OK, sorry about leading you down the garden path on the git committer.

> The issue 695 talks about having a different commit author/committer for user requested actions

I have asked for clarification in that issue.